src/main/java/jchess/gamelogic/Game.java

```diff
@@ -147,28 +169,34 @@ public void onBoardActivity(IPositionAgent oPosition) {
 			deselectedActivePosition();
 		}		
 	}
+	/**
+	 * Check Rule
+	 * Check if the King is in Check position
+	 * Save the name of the Player check and the Player in check
+	 * @return
+	 */
 	public boolean tryCheckRule()
 	{
 		//Get Active Board and Active Player
-		IBoardAgent oCurrentBoard =  this.m_oGameModel.getBoard();
-		IPlayerAgent oCurrentPlayer = this.m_oGameState.getActivePlayer();
+		IBoardAgent oCurrentBoard =  m_oGameModel.getBoard();
+		IPlayerAgent oCurrentPlayer = m_oGameState.getActivePlayer();
 		//Get the King's position of Active Player
 		String[] KingArray= new String[] {"Black", "White", "Red"};
-		String KingName = "";
+		String stKingName = "";
 		IPieceAgent oKingPiece = null;
 		IPositionAgent oKingPosition = null;
-		int kingRank = 0;
-		int kingFile = 0;
-		for(String KingColour : KingArray)
+		int nKingRank = 0;
```

Comment:
The following lines, till 201, are just getting King's piece. All these lines should go to PlayerAgent class, there you should create a function, e.g. GetKingPiece()